plugins/jobs/src/js/JobCreateEditFormModal.tsx

```diff
@@ -0,0 +1,57 @@
+import * as React from "react";
+
+import { BehaviorSubject } from "rxjs/BehaviorSubject";
+import { combineLatest } from "rxjs/observable/combineLatest";
+
+import { componentFromStream } from "data-service";
+
+import { createJob, JobData, updateJob } from "#SRC/js/events/MetronomeClient";
+import Job from "#SRC/js/structs/Job";
+
+import JobFormModal, { ErrorMessage } from "./components/JobFormModal";
+
+interface JobCreateEditFormModalProps {
+  isEdit?: boolean;
+  job?: Job;
+  open: boolean;
+  onClose: () => void;
+}
+
+type JobFormModalInput = [JobCreateEditFormModalProps, ErrorMessage];
+
+const JobCreateEditFormModal = componentFromStream<JobCreateEditFormModalProps>(
+  props$ => {
+    const errorMessage$ = new BehaviorSubject<ErrorMessage>(null);
+
+    function handleErrorMessageChange(errorMessage: ErrorMessage) {
```

Comment:
What I would like to think about is how we could use recompose `createEvenHandler` so that we operate solely with streams to get rid of these custom handlers

Reply:
I did and discussed with @Fabs. Decision was not to proceed with it at this stage.